src/xrpld/ledger/detail/ApplyStateTable.cpp

```diff
@@ -147,7 +159,8 @@ ApplyStateTable::apply(
             }
             auto const origNode = to.read(keylet::unchecked(item.first));
             auto curNode = item.second.second;
-            if ((type == &sfModifiedNode) && (*curNode == *origNode))
+            if ((type == &sfModifiedNode) && (*curNode == *origNode) &&
+                !isBatch)
                 continue;
             std::uint16_t nodeType = curNode
```

Comment:
can a batch transaction ever modify an object other than AccountRoot? if not, would be use good to assert that here.

Reply:
No I dont believe so. I will add that here.